client/src/pages/Messages/Messaging/MessagingContainer.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useState, useEffect } from 'react';
+import useStyles from '../useStyles';
+import { useAuth } from '../../../context/useAuthContext';
+import { User } from '../../../interface/User';
+import { Message } from '../../../interface/Message';
+import MessageList from './MessageList';
+import fetchUser from '../../../helpers/APICalls/getUserById';
+import fetchLastMessage from '../../../helpers/APICalls/getMessagesByConvoId';
+import pushNewMessage from '../../../helpers/APICalls/createNewMessage';
+import { Conversation } from '../../../interface/Conversation';
+import profileAvatar from '../../../Images/user.png';
+import { Typography, Grid, Divider, Avatar, Badge, Button, TextField, CircularProgress } from '@material-ui/core';
+
+interface Props {
+  convo: Conversation;
+}
+
+export default function MessagingContainer({ convo }: Props): JSX.Element {
+  const classes = useStyles();
+  const { loggedInUser } = useAuth();
+
+  const [newMessage, setNewMessage] = useState('');
+  const [isSubmitting, setSubmitting] = useState(false);
+  const [participant, setParticipant] = useState<User | null>(null);
+  const [messages, setMessages] = useState<Message[]>([]);
+
+  const participantId = convo?.participants.find((participant: string) => participant !== loggedInUser?.id);
+
+  useEffect(() => {
+    let active = true;
+
+    if (participantId) {
+      const getParticipantInfo = async () => {
```

Comment:
It looks like we are fetching this information twice, once in this component and another time in the one above. Maybe consider the usage of a context to store messages and conversations?

Reply:
I was thinking about the same thing. But cant seem to figure out how to do this architecture wise. Since we need to update the messages within each conversation as well. Will it require a major restructure of current components to support that type of data flow. Please point me in the right direction. Thank you

Reply:
implemented context api and loaded all data within context.